### DISCUSSION TITLE
Datatable stateStorage

### DISCUSSION THREAD

**Discussion by jtoon86** [primefaces/community#300] [PrimeReact] [answered]
I am attempting to save the data table state to local storage so my users can setup their filters/etc and have it persist between visits to the site.

On the DataTable component, I set stateStorage="local" and stateKey="my_form" .. looking at the local storage, I do see the my_form being created and updated as the state of DataTable is adjusted. GREAT!

However -- As soon as I reload the page, my local storage gets reset to {multiSortMeta:[]}

Below is a simple example showing the issue. 

```jsx
import { DataTable } from "primereact/datatable";
import { Column } from "primereact/column";

export default function SimpleForm() {
    return (
        <DataTable stateStorage="local" stateKey="my_form">
            <Column key="col1" field="col1" header="Col1" filter></Column> 
            <Column key="col2" field="col2" header="Col2" filter></Column> 
        </DataTable>
    )
}
```

**Comment by melloware** [comment, marked as answer]
Most likely a bug.  I would open a ticket with your StackBlitz reproducer.

**Reply by jtoon86** [reply]
Thanks. I went ahead and created [ticket #5159](https://github.com/primefaces/primereact/issues/5159).
